Kconfig.zephyr

```diff
@@ -162,6 +162,12 @@ config CMAKE_LINKER_GENERATOR
 
 endchoice
 
+config LLVM_USE_BUILTIN_LLD
+	bool "LLVM uses built-in lld linker"
```

Comment:
Maybe we should consider `lld` to be the linker for `llvm`, and then have a setting requesting `ld` to be used as linker.

We can still default `llvm` to use `ld`, as we do today.

For example:
```
config LLVM_USE_LD
	bool "LLVM uses ld linker"
	default y
	help
	  Use binutils ld linker instead of LLVM built-in lld linker.
```

Reply:
thanks, have updated it using your suggestions.